org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/InstallationManagement.scala

```diff
@@ -158,13 +160,22 @@ trait InstallationManagement { this: ScalaProject =>
     projectSpecificStorage.addPropertyChangeListener(compilerSettingsListener)
   }
 
+  private lazy val actualAdditionalParams: AtomicReference[() => String] = new AtomicReference(() => {
+    actualAdditionalParams.getAndSet(() => storage.getString(CompilerSettings.ADDITIONAL_PARAMS))
```

Comment:
There is actually some documentation regarding thread safety for `set`, though it is placed in the package documentation: http://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/package-summary.html.